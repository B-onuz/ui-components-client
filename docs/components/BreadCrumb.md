# BreadCrumb (Component)

Componente de Breadcrumbs

## Como utilizar

```js
import BreadCrumb from 'ui-components/lib/components/BreadCrumb'
```

```jsx
import React from 'react'
import BreadCrumb from './BreadCrumb'
import { Flex, Box } from 'reflexbox'

const exampleObj = [
  {
    label: 'Gestão da rota do Link 1',
    onClick: () => ({}),
  },
  {
    label: 'Gestão da rota do Link 2',
    onClick: () => ({}),
  },
  {
    label: 'Gestão da rota do Link 3',
    onClick: () => ({}),
  },
  {
    label: 'Gestão da rota do Link 4',
    onClick: () => ({}),
  },
]

export const BreadCrumbColors = () => (
  <Flex flexDirection={'column'}>
    <Box>
      <BreadCrumb navigationLinks={exampleObj} color={'primary'} />
    </Box>
    <Box>
      <BreadCrumb navigationLinks={exampleObj} color={'violet'} />
    </Box>
    <Box>
      <BreadCrumb navigationLinks={exampleObj} color={'info'} />
    </Box>
    <Box>
      <BreadCrumb navigationLinks={exampleObj} color={'olive'} />
    </Box>
  </Flex>
)
```

## Variações

### Color

```jsx
<BreadCrumb color={'primary'} />
```

Escala no tema: `theme.color`

## Styled-system props

- space https://styled-system.com/table#space
