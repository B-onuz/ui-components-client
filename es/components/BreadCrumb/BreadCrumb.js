"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _styledSystem = require("styled-system");

var _Button = _interopRequireDefault(require("../Button"));

var _reactFontawesome = require("@fortawesome/react-fontawesome");

var _freeSolidSvgIcons = require("@fortawesome/free-solid-svg-icons");

var _color = _interopRequireDefault(require("color"));

var _this = void 0,
    _jsxFileName = "/Users/master/projects/bonuz/ui-components/src/components/BreadCrumb/BreadCrumb.js";

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n  ", "\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n  color: ", ";\n  padding: 8px 16px;\n  font-size: 14px;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  display: flex;\n  align-items: center;\n  flex-wrap: wrap;\n  color: ", ";\n  ", "\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var StyledBreadCrumb = _styledComponents["default"].div(_templateObject(), function (_ref) {
  var theme = _ref.theme,
      color = _ref.color;
  return "" + theme.colors[color];
}, _styledSystem.space);

var ActiveRoute = _styledComponents["default"].span(_templateObject2(), function (_ref2) {
  var theme = _ref2.theme,
      color = _ref2.color;
  return "" + theme.colors[color];
});

var IconColor = function IconColor(_ref3) {
  var theme = _ref3.theme,
      color = _ref3.color;
  var newColor = theme.colors[color];
  var darkColor = new _color["default"](newColor);
  var navIconColor = darkColor.darken(0.5).toString();
  return "\n    color: " + navIconColor + "\n  ";
};

var Icon = (0, _styledComponents["default"])(_reactFontawesome.FontAwesomeIcon)(_templateObject3(), IconColor);

var BreadCrumb = function BreadCrumb(_ref4) {
  var children = _ref4.children,
      navigationLinks = _ref4.navigationLinks,
      color = _ref4.color,
      rest = _objectWithoutPropertiesLoose(_ref4, ["children", "navigationLinks", "color"]);

  return /*#__PURE__*/_react["default"].createElement(StyledBreadCrumb, _extends({}, rest, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 39,
      columnNumber: 5
    }
  }), (navigationLinks || []).map(function (item, index) {
    var label = item.label,
        key = item.key,
        itemProps = _objectWithoutPropertiesLoose(item, ["label", "key"]);

    return index !== navigationLinks.length - 1 ? /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, {
      key: index,
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 43,
        columnNumber: 11
      }
    }, /*#__PURE__*/_react["default"].createElement(_Button["default"], _extends({
      key: key || index,
      linkButton: true,
      size: "small",
      color: color
    }, itemProps, {
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 44,
        columnNumber: 13
      }
    }), label), /*#__PURE__*/_react["default"].createElement(Icon, {
      icon: _freeSolidSvgIcons.faAngleRight,
      color: color,
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 47,
        columnNumber: 13
      }
    })) : /*#__PURE__*/_react["default"].createElement(ActiveRoute, _extends({
      key: key || index,
      color: 'black'
    }, itemProps, {
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 50,
        columnNumber: 11
      }
    }), item.label);
  }));
};

BreadCrumb.defaultProps = {
  color: 'primary'
};
var _default = BreadCrumb;
exports["default"] = _default;