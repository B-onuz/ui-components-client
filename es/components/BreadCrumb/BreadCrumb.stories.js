'use strict'

Object.defineProperty(exports, '__esModule', {
  value: true,
})
exports.LongListItems = exports.BreadCrumbColors = exports['default'] = void 0

var _react = _interopRequireDefault(require('react'))

var _BreadCrumb = _interopRequireDefault(require('./BreadCrumb'))

var _reflexbox = require('reflexbox')

var _addonActions = require('@storybook/addon-actions')

var _this = void 0,
  _jsxFileName = '/home/junior/Fontes/ui-components-client/src/components/BreadCrumb/BreadCrumb.stories.js'

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj }
}

var _default = {
  title: 'BreadCrumb',
  component: _BreadCrumb['default'],
}
exports['default'] = _default
var longExampleObj = [
  {
    label: 'Gestão da rota do Link 1',
    onClick: (0, _addonActions.action)('Ação do Link'),
  },
  {
    label: 'Gestão da rota do Link 2',
    onClick: (0, _addonActions.action)('Ação do Link'),
  },
  {
    label: 'Gestão da rota do Link 3',
    onClick: (0, _addonActions.action)('Ação do Link'),
  },
  {
    label: 'Gestão da rota do Link 4',
    onClick: (0, _addonActions.action)('Ação do Link'),
  },
  {
    label: 'Gestão da rota do Link 5',
    onClick: (0, _addonActions.action)('Ação do Link'),
  },
  {
    label: 'Gestão da rota do Link 6',
    onClick: (0, _addonActions.action)('Ação do Link'),
  },
  {
    label: 'Gestão da rota do Link 7',
    onClick: (0, _addonActions.action)('Ação do Link'),
  },
  {
    label: 'Gestão da rota do Link 8',
    onClick: (0, _addonActions.action)('Ação do Link'),
  },
  {
    label: 'Gestão da rota do Link 9',
    onClick: (0, _addonActions.action)('Ação do Link'),
  },
  {
    label: 'Gestão da rota do Link 10',
    onClick: (0, _addonActions.action)('Ação do Link'),
  },
  {
    label: 'Gestão da rota do Link 11',
    onClick: (0, _addonActions.action)('Ação do Link'),
  },
  {
    label: 'Gestão da rota do Link 12',
    onClick: (0, _addonActions.action)('Ação do Link'),
  },
]
var exampleObj = [
  {
    label: 'Gestão da rota do Link 1',
    onClick: (0, _addonActions.action)('Ação do Link'),
  },
  {
    label: 'Gestão da rota do Link 2',
    onClick: (0, _addonActions.action)('Ação do Link'),
  },
  {
    label: 'Gestão da rota do Link 3',
    onClick: (0, _addonActions.action)('Ação do Link'),
  },
  {
    label: 'Gestão da rota do Link 4',
    onClick: (0, _addonActions.action)('Ação do Link'),
  },
]

var BreadCrumbColors = function BreadCrumbColors() {
  return /*#__PURE__*/ _react['default'].createElement(
    _reflexbox.Flex,
    {
      flexDirection: 'column',
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 82,
        columnNumber: 3,
      },
    },
    /*#__PURE__*/ _react['default'].createElement(
      _reflexbox.Box,
      {
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 83,
          columnNumber: 5,
        },
      },
      /*#__PURE__*/ _react['default'].createElement(_BreadCrumb['default'], {
        navigationLinks: exampleObj,
        color: 'primary',
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 84,
          columnNumber: 7,
        },
      })
    ),
    /*#__PURE__*/ _react['default'].createElement(
      _reflexbox.Box,
      {
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 86,
          columnNumber: 5,
        },
      },
      /*#__PURE__*/ _react['default'].createElement(_BreadCrumb['default'], {
        navigationLinks: exampleObj,
        color: 'violet',
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 87,
          columnNumber: 7,
        },
      })
    ),
    /*#__PURE__*/ _react['default'].createElement(
      _reflexbox.Box,
      {
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 89,
          columnNumber: 5,
        },
      },
      /*#__PURE__*/ _react['default'].createElement(_BreadCrumb['default'], {
        navigationLinks: exampleObj,
        color: 'info',
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 90,
          columnNumber: 7,
        },
      })
    ),
    /*#__PURE__*/ _react['default'].createElement(
      _reflexbox.Box,
      {
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 92,
          columnNumber: 5,
        },
      },
      /*#__PURE__*/ _react['default'].createElement(_BreadCrumb['default'], {
        navigationLinks: exampleObj,
        color: 'olive',
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 93,
          columnNumber: 7,
        },
      })
    )
  )
}

exports.BreadCrumbColors = BreadCrumbColors

var LongListItems = function LongListItems() {
  return /*#__PURE__*/ _react['default'].createElement(_BreadCrumb['default'], {
    navigationLinks: longExampleObj,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 98,
      columnNumber: 36,
    },
  })
}

exports.LongListItems = LongListItems
