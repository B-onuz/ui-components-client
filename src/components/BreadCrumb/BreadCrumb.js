import React from 'react'
import styled from 'styled-components'
import { color } from 'styled-system'
import { space } from 'styled-system'
import Button from '../Button'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faAngleRight } from '@fortawesome/free-solid-svg-icons'
import Color from 'color'

const StyledBreadCrumb = styled.div`
  display: flex;
  align-items: center;
  flex-wrap: wrap;
  color: ${({ theme, color }) => `${theme.colors[color]}`};
  ${space}
`

const ActiveRoute = styled.span`
  color: ${({ theme, color }) => `${theme.colors[color]}`};
  padding: 8px 16px;
  font-size: 14px;
`

const IconColor = ({ theme, color }) => {
  const newColor = theme.colors[color]
  const darkColor = new Color(newColor)
  const navIconColor = darkColor.darken(0.5).toString()
  return `
    color: ${navIconColor}
  `
}

const Icon = styled(FontAwesomeIcon)`
  ${IconColor}
`

const BreadCrumb = ({ children, navigationLinks, color, ...rest }) => {
  return (
    <StyledBreadCrumb {...rest}>
      {(navigationLinks || []).map((item, index) => {
        const { label, key, ...itemProps } = item
        return index !== navigationLinks.length - 1 ? (
          <React.Fragment key={index}>
            <Button key={key || index} linkButton size="small" color={color} {...itemProps}>
              {label}
            </Button>
            <Icon icon={faAngleRight} color={color} />
          </React.Fragment>
        ) : (
          <ActiveRoute key={key || index} color={'black'} {...itemProps}>
            {item.label}
          </ActiveRoute>
        )
      })}
    </StyledBreadCrumb>
  )
}

BreadCrumb.defaultProps = {
  color: 'primary',
}

export default BreadCrumb
