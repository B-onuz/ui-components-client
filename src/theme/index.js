const colors = {
  red: '#db2828',
  lightRed: '#ef4d5a',
  mediumRed: '#FF4F5D',
  darkRed: '#9c2543',
  fullDarkRed: '#550e33',
  orange: '#f2711c',
  yellow: '#fbbd08',
  olive: '#b5cc18',
  green: '#21BA45',
  teal: '#00b5ad',
  blue: '#2185d0',
  facebookBlue: '#3B5998',
  blueLink: '#4285F4',
  violet: '#6435c9',
  purple: '#a333c8',
  pink: '#e03997',
  brown: '#a5673f',
  grey: '#767676',
  textGrey: '#8D8D8D',
  lightGrey: '#6D6D6D',
  mediumGrey: '#838383',
  darkGrey: '#9A9A9A',
  boldGray: '#55555A',
  black: '#1b1c1d',
  white: '#fdfdfd',
  warning: '#7A4D05',
  success: '#1E561F',
  info: '#0E566C',
  danger: '#973937',
  cyan: '#1CB5AD',
  disabledGrey: '#ffffff1f',
  disabledGreyText: '#ffffff4d',
}

colors.primary = colors.lightRed
colors.secondary = colors.fullDarkRed
colors.default = '#e0e1e2'

const breakpoints = ['40em', '52em', '64em', '80em']

breakpoints.sm = breakpoints[0]
breakpoints.md = breakpoints[1]
breakpoints.lg = breakpoints[2]
breakpoints.xl = breakpoints[3]

const fontSizes = [12, 14, 16, 20, 24, 32]

export default {
  fontFamily: `'Lato', sans-serif`,
  colors,
  fontSizes,
  breakpoints,
}
