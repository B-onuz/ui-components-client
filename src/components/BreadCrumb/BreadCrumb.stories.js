import React from 'react'
import BreadCrumb from './BreadCrumb'
import { Flex, Box } from 'reflexbox'
import { action } from '@storybook/addon-actions'

export default {
  title: 'BreadCrumb',
  component: BreadCrumb,
}

const longExampleObj = [
  {
    label: 'Gestão da rota do Link 1',
    onClick: action('Ação do Link'),
  },
  {
    label: 'Gestão da rota do Link 2',
    onClick: action('Ação do Link'),
  },
  {
    label: 'Gestão da rota do Link 3',
    onClick: action('Ação do Link'),
  },
  {
    label: 'Gestão da rota do Link 4',
    onClick: action('Ação do Link'),
  },
  {
    label: 'Gestão da rota do Link 5',
    onClick: action('Ação do Link'),
  },
  {
    label: 'Gestão da rota do Link 6',
    onClick: action('Ação do Link'),
  },
  {
    label: 'Gestão da rota do Link 7',
    onClick: action('Ação do Link'),
  },
  {
    label: 'Gestão da rota do Link 8',
    onClick: action('Ação do Link'),
  },
  {
    label: 'Gestão da rota do Link 9',
    onClick: action('Ação do Link'),
  },
  {
    label: 'Gestão da rota do Link 10',
    onClick: action('Ação do Link'),
  },
  {
    label: 'Gestão da rota do Link 11',
    onClick: action('Ação do Link'),
  },
  {
    label: 'Gestão da rota do Link 12',
    onClick: action('Ação do Link'),
  },
]

const exampleObj = [
  {
    label: 'Gestão da rota do Link 1',
    onClick: action('Ação do Link'),
  },
  {
    label: 'Gestão da rota do Link 2',
    onClick: action('Ação do Link'),
  },
  {
    label: 'Gestão da rota do Link 3',
    onClick: action('Ação do Link'),
  },
  {
    label: 'Gestão da rota do Link 4',
    onClick: action('Ação do Link'),
  },
]

export const BreadCrumbColors = () => (
  <Flex flexDirection={'column'}>
    <Box>
      <BreadCrumb navigationLinks={exampleObj} color={'primary'} />
    </Box>
    <Box>
      <BreadCrumb navigationLinks={exampleObj} color={'violet'} />
    </Box>
    <Box>
      <BreadCrumb navigationLinks={exampleObj} color={'info'} />
    </Box>
    <Box>
      <BreadCrumb navigationLinks={exampleObj} color={'olive'} />
    </Box>
  </Flex>
)

export const LongListItems = () => <BreadCrumb navigationLinks={longExampleObj} />
