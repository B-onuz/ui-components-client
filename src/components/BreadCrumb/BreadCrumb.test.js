import React from 'react'
import BreadCrumb from './BreadCrumb.js'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import theme from '../../theme'
import 'jest-styled-components'

test('Should render BreadCrumb', () => {
  const exampleObj = [
    {
      label: 'Gestão da rota do Link 1',
      action: () => console.log('Ação do Link'),
    },
    {
      label: 'Gestão da rota do Link 2',
      action: () => console.log('Ação do Link'),
    },
    {
      label: 'Gestão da rota do Link 3',
      action: () => console.log('Ação do Link'),
    },
    {
      label: 'Gestão da rota do Link 4',
      action: () => console.log('Ação do Link'),
    },
  ]

  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <BreadCrumb navigationLinks={exampleObj} />
    </ThemeProvider>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Should render BreadCrumb with color', () => {
  const exampleObj = [
    {
      label: 'Gestão da rota do Link 1',
      action: () => console.log('Ação do Link'),
    },
    {
      label: 'Gestão da rota do Link 2',
      action: () => console.log('Ação do Link'),
    },
    {
      label: 'Gestão da rota do Link 3',
      action: () => console.log('Ação do Link'),
    },
    {
      label: 'Gestão da rota do Link 4',
      action: () => console.log('Ação do Link'),
    },
  ]

  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <BreadCrumb navigationLinks={exampleObj} color={'violet'} />
    </ThemeProvider>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
