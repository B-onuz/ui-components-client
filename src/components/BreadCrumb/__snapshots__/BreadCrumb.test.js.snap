// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render BreadCrumb 1`] = `
.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font-family: inherit;
  border-radius: 4px;
  outline: none;
  cursor: pointer;
  padding: .5rem 1rem;
  border-style: solid;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  border-color: undefined !important;
  color: #fff;
  background-color: #ef4d5a;
  border-color: #ef4d5a;
  font-size: 14px;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #ef4d5a;
  border: none;
  background-color: transparent !important;
  box-shadow: none !important;
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1:focus {
  box-shadow: 0 0 8px #ef4d5a;
}

.c1:active:not(:disabled) {
  box-shadow: inset 0px 0px 5px #c1c1c1;
}

.c1:active,
.c1:hover {
  background-color: hsl(355.20000000000005,91.9%,55.8%);
}

.c1:disabled {
  background-color: hsl(355.20000000000005,50.1%,68.2%);
}

.c1:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  background-color: transparent;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  color: undefined;
}

.c3 {
  color: #1b1c1d;
  padding: 8px 16px;
  font-size: 14px;
}

.c2 {
  color: hsl(355.20000000000005,83.5%,31%);
}

<div
  className="c0"
>
  <button
    action={[Function]}
    className="c1"
    color="primary"
    disabled={false}
    size="small"
    type="button"
  >
    Gestão da rota do Link 1
  </button>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-angle-right fa-w-8 c2"
    color="primary"
    data-icon="angle-right"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 256 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <button
    action={[Function]}
    className="c1"
    color="primary"
    disabled={false}
    size="small"
    type="button"
  >
    Gestão da rota do Link 2
  </button>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-angle-right fa-w-8 c2"
    color="primary"
    data-icon="angle-right"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 256 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <button
    action={[Function]}
    className="c1"
    color="primary"
    disabled={false}
    size="small"
    type="button"
  >
    Gestão da rota do Link 3
  </button>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-angle-right fa-w-8 c2"
    color="primary"
    data-icon="angle-right"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 256 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <span
    action={[Function]}
    className="c3"
    color="black"
  >
    Gestão da rota do Link 4
  </span>
</div>
`;

exports[`Should render BreadCrumb with color 1`] = `
.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font-family: inherit;
  border-radius: 4px;
  outline: none;
  cursor: pointer;
  padding: .5rem 1rem;
  border-style: solid;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  border-color: undefined !important;
  color: #fff;
  background-color: #6435c9;
  border-color: #6435c9;
  font-size: 14px;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #6435c9;
  border: none;
  background-color: transparent !important;
  box-shadow: none !important;
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1:focus {
  box-shadow: 0 0 8px #6435c9;
}

.c1:active:not(:disabled) {
  box-shadow: inset 0px 0px 5px #c1c1c1;
}

.c1:active,
.c1:hover {
  background-color: hsl(259.1,64.1%,44.8%);
}

.c1:disabled {
  background-color: hsl(259.1,35%,54.8%);
}

.c1:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  background-color: transparent;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  color: undefined;
}

.c3 {
  color: #1b1c1d;
  padding: 8px 16px;
  font-size: 14px;
}

.c2 {
  color: hsl(259.1,58.3%,24.9%);
}

<div
  className="c0"
>
  <button
    action={[Function]}
    className="c1"
    color="violet"
    disabled={false}
    size="small"
    type="button"
  >
    Gestão da rota do Link 1
  </button>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-angle-right fa-w-8 c2"
    color="violet"
    data-icon="angle-right"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 256 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <button
    action={[Function]}
    className="c1"
    color="violet"
    disabled={false}
    size="small"
    type="button"
  >
    Gestão da rota do Link 2
  </button>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-angle-right fa-w-8 c2"
    color="violet"
    data-icon="angle-right"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 256 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <button
    action={[Function]}
    className="c1"
    color="violet"
    disabled={false}
    size="small"
    type="button"
  >
    Gestão da rota do Link 3
  </button>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-angle-right fa-w-8 c2"
    color="violet"
    data-icon="angle-right"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 256 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <span
    action={[Function]}
    className="c3"
    color="black"
  >
    Gestão da rota do Link 4
  </span>
</div>
`;
